### PR TITLE
first draft of drivers package

### PR DESCRIPTION
**Pull request by mmoskal**
This is completely untested, probably doesn't compile.

### CODE REVIEW

File: packages/drivers/src/servers.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    implements ds.ITemperatureServer
+{
+    constructor(public options: TemperatureOptions) {
+        super(ds.Temperature.spec, "temperature")
+    }
+
+    async temperature() {
+        return await this.options.read()
+    }
+    minTemperature() {
```
Comment (pelikhan) [inline packages/drivers/src/servers.ts:22]: min -> Temperature
temperature -> Error

should we always order things with name + suffix?

File: packages/drivers/src/servers.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+export interface TemperatureOptions {
+    min: number
+    max: number
+    error: number
+    read: () => Promise<number>
+}
+
+export class TemperatureServer
+    extends SensorServer<ds.ITemperatureServer>
+    implements ds.ITemperatureServer
```
Comment (pelikhan) [inline packages/drivers/src/servers.ts:13]: TypeScript typically does not use the `I*` naming convention, it could just be `ds.TemperatureServer`

**Review by pelikhan — APPROVED** [review]

File: packages/drivers/src/servers.ts (hunk 2 above)
Comment (mmoskal) [reply packages/drivers/src/servers.ts:13]: yeah, but then we have `TemperatureServer` and `ds.TemperatureServer`

File: packages/drivers/src/servers.ts (hunk 1 above)
Comment (mmoskal) [reply packages/drivers/src/servers.ts:22]: I don't understand; the order and names come from the spec

File: packages/drivers/src/servers.ts (hunk 2 above)
Comment (pelikhan) [reply packages/drivers/src/servers.ts:13]: `ds.TemperatureServerSpec` ?

Comment (mmoskal) [reply packages/drivers/src/servers.ts:13]: Will change

**Pull request merged**